app/subscribers/publish_zync_event_subscriber.rb

```diff
@@ -10,25 +10,43 @@ def initialize(publisher = DEFAULT_PUBLISHER)
 
   attr_reader :publisher
 
-  # @param [ZyncEvent] event
-  def call(event)
-    unless ThreeScale.config.onpremises
+  delegate :domain_event_in_saas?, :build_zync_event, to: :class
+
+  class << self
+    def domain_event_in_saas?(event)
+      ThreeScale.saas? && (event.is_a?(Domains::ProxyDomainsChangedEvent) || event.is_a?(Domains::ProviderDomainsChangedEvent))
+    end
+
+    def build_zync_event(event)
       case event
-      when Domains::ProxyDomainsChangedEvent, Domains::ProviderDomainsChangedEvent
-        return
+      when ApplicationRelatedEvent
+        zync_metadata = event.metadata.fetch(:zync, {})
+        zync_metadata[:oidc_auth_enabled] || sync_non_oidc_apps ? ZyncEvent.create(event, event.application) : nil
```

Comment:
instead of adding `oidc_auth_enabled: service.oauth?` to all events, isn't it better to keep the mas before but have a check like
```suggestion
        event.metadata.fetch[:service]&.oauth? || sync_non_oidc_apps ? ZyncEvent.create(event, event.application) : nil
```
or optimize a little

```suggestion
        sync_non_oidc_apps || event.metadata.fetch[:service]&.oauth? ? ZyncEvent.create(event, event.application) : nil
```

Reply:
Well, actually, only `Applications::ApplicationCreatedEvent` has the service object in its body, the others have only the service ID in `metadata.zync.service_id`, I don't think it's worth initializing the object just for that. Also, not sure it is worth passing the object around.
But I don't know, if you strongly prefer to use the whole `service` instead `sync_non_oidc_apps`, I can change it.

Reply:
Unfortunately, changing `service_id` for `service` inside the event data breaks the scenario `test_create_and_publish_when_the_application_associations_do_not_exist_anymore`

So, all the events are stored in the DB, and are serialized with GlobalID. For deleted events, an example of `data` field would be:
```
---
application:
  _aj_globalid: gid://system/MissingModel::MissingApplication/564
service_backend_id: '2'
application_id: cf699eed
service:
  _aj_globalid: gid://system/Service/2
_aj_symbol_keys:
- application
- service_backend_id
- application_id
- service
```

So, if the service is already deleted, then the event will fail to deserialize...